doc/changelog/01-kernel/19262-template-no-sup-constraint-kernel.rst

```diff
@@ -0,0 +1,9 @@
+- **Changed:**
+  the kernel typing rules for template polymorphic inductive types do not
+  require anymore adding global constraints when applied enough. Rather,
+  template polymorphic inductive types are now a special kind of universe
+  polymorphic inductive types that do not need explicit instances and
+  can handle some amount of algebraic universe levels. The new rules are
+  strictly more general than the previous ones and thus backwards compatible
```

Comment:
Should we have another changelog for the new restrictions?

Reply:
Rather than a changelog, which is easily lost in the mist of time, I think that there should be some documentation in the refman about template polymorphism.

Reply:
I'd rather have both, then if 2 years later someone asks about some code that broke because of those changes it will be easier to understand what happened instead of staring at a changelog which claims to be backwards compatible.

Reply:
I've pushed a minimal changelog included in this PR. Feel free to expand the refman.